src/modules/errors/components/ApolloErrorAlert.tsx

```diff
@@ -8,40 +8,61 @@ import {
   Snackbar,
 } from '@mui/material';
 
-import { GraphQLError } from 'graphql';
 import { forwardRef, useEffect, useMemo, useState } from 'react';
 
 import { isServerError, UNKNOWN_ERROR_HEADING } from '../util';
 import ApolloErrorTrace from './ApolloErrorTrace';
 
 interface Props2 {
-  graphqlErrors: GraphQLError[];
+  errors: Error[];
+  isNetworkError: boolean;
   alertProps?: AlertProps;
   retry?: VoidFunction;
 }
 
 const BaseAlert = forwardRef<Props2, any>(
-  ({ alertProps, retry, graphqlErrors }, ref) => {
+  ({ alertProps, retry, errors, isNetworkError }, ref) => {
+    let errorMessage = '';
+    if (isNetworkError) {
+      errorMessage = `There was a problem connecting to the network. Please reload the page and try again.`;
+    } else {
+      errorMessage = errors[0]?.message;
+      // replace terse unknown error message with something more helpful
+      if (!errorMessage || errorMessage == UNKNOWN_ERROR_HEADING) {
```

Comment:
This is dependent on the fact that `UNKNOWN_ERROR_HEADING` == [INTERNAL_ERROR_DISPLAY_MESSAGE on the backend](https://github.com/greenriver/hmis-warehouse/blob/58843a72e2db8b1e864ba253d9b0c9bb8d692d32/drivers/hmis/lib/hmis_errors/api_error.rb#L10)?

I was thinking that the backend itself would return the user-facing error message, which is what happens currently. Feels a bit shaky to rely on the strings being equal.

Reply:
definitely, I'll pull this into the backend for now. 

Capturing our slack convo: in the future we might consider giving the error object a type and allow the FE to handle the display